semgrep-monitor

```diff
@@ -73,13 +130,57 @@ def main():
         "username": "Semgrep",
     }
     try:
-        urllib.request.urlopen(
-            args.slack_url,
-            urllib.parse.urlencode({"payload": json.dumps(payload)}).encode(),
+        response = requests.post(args.slack_url, data={"payload": json.dumps(payload)},)
+        response.raise_for_status()
+    except requests.RequestException:
+        print(f"Slack returned this error: {response.text}", file=sys.stderr)
+
+
+def report_finish_to_semgrep_app():
+    if not args.semgrep_app_token:
+        return
+
+    # send findings
+    results = json.load(args.results_file)
+    payload = [
+        {
+            "path": result["path"],
+            "rule": result["check_id"],
+            "startLine": result["line"],
+            "endLine": result["line"],  # TODO get end line from semgrep
+        }
+        for result in results
+    ]
+    api_url = f"{args.semgrep_app_url}/api/agent/scan/{args.scan_id}/findings"
+    try:
+        response = requests.post(
+            api_url,
+            headers={"Authorization": f"Bearer {args.semgrep_app_token}"},
+            json=payload,
         )
-    except urllib.error.HTTPError as e:
-        print(f"Slack returned this error: {e.read().decode()}")
+        response.raise_for_status()
+    except requests.RequestException:
+        print(f"Semgrep App returned this error: {response.text}", file=sys.stderr)
+
+    # mark as complete
+    stats = {
+        "findings": len(results)
```

Comment:
You can put whatever stats you want here; the API example ones were just random ones I came up with in 10 seconds of brainstorming.